lua/neo-tree/setup/init.lua

```diff
@@ -307,41 +307,45 @@ M.merge_config = function(config, is_auto_config)
   -- setup the default values for all sources
   normalize_mappings(default_config)
   for _, source_name in ipairs(sources) do
-    local source_config = default_config[source_name]
+    local source_default_config = default_config[source_name]
     local mod_root = "neo-tree.sources." .. source_name
-    source_config.components = require(mod_root .. ".components")
-    source_config.commands = require(mod_root .. ".commands")
-    source_config.name = source_name
+    source_default_config.components = require(mod_root .. ".components")
+    source_default_config.commands = require(mod_root .. ".commands")
+    source_default_config.name = source_name
 
     -- Make sure all the mappings are normalized so they will merge properly.
-    normalize_mappings(source_config)
+    normalize_mappings(source_default_config)
     normalize_mappings(config[source_name])
 
-    -- merge the global config with the source specific config
-    source_config.window = vim.tbl_deep_extend(
-      "force",
-      default_config.window or {},
-      source_config.window or {},
-      config.window or {}
-    )
-    source_config.renderers = source_config.renderers or {}
+    if config.use_default_mappings then
```

Comment:
Now that it moves in the opposite direction, it fails the tests because `config.use_default_mappings` is `nil`.

You can run `make test` from the root of the repo to run them.

Reply:
> You can run make test from the root of the repo to run the tests.

I'm not sure where to put the `plenary.nvim` and `nui.nvim` before running the tests.

I get this:

```
TEST WITH BLANK CONFIG
nvim --headless --noplugin -u tests/mininit.lua -c "lua require('plenary.test_harness').test_directory('tests/neo-tree/', {minimal_init='tests/mininit.lua',sequential=true})"
Error detected while processing /home/shaikat/.local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:
E5113: Error while calling lua chunk: ...local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:8: Vim(packadd):E919: Directory not found in 'packpath': "pack/*/opt/nui.nvim"
stack traceback:
        [C]: in function 'cmd'
        ...local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:8: in main chunk^[lStarting...Scheduling: tests/neo-tree/command/command_spec.lua
^[l^[l^[lScheduling: tests/neo-tree/command/command_current_spec.lua
Error detected while processing /home/shaikat/.local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:
E5113: Error while calling lua chunk: ...local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:8: Vim(packadd):E919: Directory not found in 'packpath': "pack/*/opt/nui.nvim"
stack traceback:
        [C]: in function 'cmd'
        ...local/share/nvim/plugged/neo-tree.nvim/tests/mininit.lua:8: in main chunk

========================================
Testing:        /home/shaikat/.local/share/nvim/plugged/neo-tree.nvim/tests/neo-tree/command/command_spec.lua
========================================
FAILED TO LOAD FILE
...nvim/plugged/neo-tree.nvim//lua/neo-tree/ui/renderer.lua:2: module 'nui.line' not found:
```

But I tried to fix the issue and pushed it.